src/mbgl/renderer/painter_fill_extrusion.cpp

```diff
@@ -0,0 +1,88 @@
+#include <mbgl/renderer/painter.hpp>
+#include <mbgl/renderer/paint_parameters.hpp>
+#include <mbgl/renderer/fill_extrusion_bucket.hpp>
+#include <mbgl/renderer/render_tile.hpp>
+#include <mbgl/style/layers/fill_extrusion_layer.hpp>
+#include <mbgl/style/layers/fill_extrusion_layer_impl.hpp>
+#include <mbgl/sprite/sprite_atlas.hpp>
+#include <mbgl/programs/programs.hpp>
+#include <mbgl/programs/fill_extrusion_program.hpp>
+#include <mbgl/util/constants.hpp>
+#include <mbgl/util/convert.hpp>
+
+namespace mbgl {
+
+using namespace style;
+
+void Painter::renderFillExtrusion(PaintParameters& parameters,
+                                  FillExtrusionBucket& bucket,
+                                  const FillExtrusionLayer& layer,
+                                  const RenderTile& tile,
+                                  const Style& style) {
```

Comment:
Rather than passing the `Style` object through the whole bucket dispatch system, can we retrieve the global light property as part of `Painter::render()` and store it as a property on the `Painter` object?